### PR TITLE
Email notifications fix (AB#90476, AB#90480, AB#90517, AB#90474, AB#90473)

### PR DESCRIPTION
**Pull request by HasnatL-Adapptlimited**
# Links
[Base PR](https://github.com/ReliefApplications/ems-frontend/pull/2460)
[back-end PR](https://github.com/ReliefApplications/ems-backend/pull/1023)

# Tickets
- [x] 90476 - Header Title changes as user types
- [x] 90480 - Email Notification Type uses ui-select menu, and disabled
- [x] 90517 - Header, Footer and Banner no longer empty
- [x] 90474 - Correct Operators showing for Resource, Resources data
- [x] 90473 - Correct Operators Showing for User and owner data

# Adappt QA bug fixes
- [x] Geospatial Operators fixed
- [x] Geospatial Data persisting (displayed as "Country Name (Lat, Long)")
- [x] **Time stamps consistent in tables across both preview screens, and in final email** -> not solved
- [x] All labels and data cell text in tables across both preview screens and final email are left alligned to be consistent.
- [x] Missing Fields Added

### CODE REVIEW

**Comment by HasnatL-Adapptlimited** [comment]
Hey @Matthis-M-ReliefApps, @AntoineRelief:

Here are the initial bug fixes as promised we will be pushing more in the next coming days via their bug link

**Comment by AntoineRelief** [comment]
[AB#90480](https://dev.azure.com/WHOHQ/2888b816-2d79-49ef-a86e-561e6cd456a0/_workitems/edit/90480) - solved

File: libs/shared/src/lib/components/email/email.service.ts [hunk 1]
```diff
@@ -18,6 +18,8 @@ import { RestService } from '../../services/rest/rest.service';
   providedIn: 'root',
 })
 export class EmailService {
+  public index = new BehaviorSubject(null);
```
Comment (AntoineRelief) [inline libs/shared/src/lib/components/email/email.service.ts:21]: @HasnatL-Adapptlimited 
I'm going to accept the changes like that for now, but I don't think edition of title should pass by the email service

**Comment by AntoineRelief** [comment]
@HasnatL-Adapptlimited 

about this ticket:
https://dev.azure.com/WHOHQ/EMSSAFE/_workitems/edit/90513
Which I believe this PR is also about

there is a difference between the dates you see in the front-end & the one sent by the back-end:

Capture d’écran 2024-04-24 à 17 31 19
Capture d’écran 2024-04-24 à 17 30 51

I don't know which format is correct so please make according changes

File: libs/shared/src/lib/components/email/filter/filter.constant.ts [hunk 2]
```diff
@@ -272,6 +272,35 @@ export const FIELD_TYPES = [
     defaultOperator: 'isnull',
     operators: ['isnull', 'isnotnull'],
   },
+  {
+    editor: 'users',
+    defaultOperator: 'contains',
+    operators: [
```
Comment (AntoineRelief) [inline libs/shared/src/lib/components/email/filter/filter.constant.ts:278]: @HasnatL-Adapptlimited 

for my information
is there any improvement planned to filtering, and if yes, for when?

here's how filtering on "owner" question looks like in a grid:
Capture d’écran 2024-04-24 à 18 00 19
here's how it looks like in the email notification:
Capture d’écran 2024-04-24 à 18 03 00

so with current status, we can't filter on these fields, as it requires a quite deep knowledge of the data structure to do so
and it's more or less the same for most of the fields

same question for the possibility to build and / or groups. Is it for second phase?

cc @Matthis-M-ReliefApps

Comment (HasnatL-Adapptlimited) [reply libs/shared/src/lib/components/email/filter/filter.constant.ts:278]: Hey @AntoineRelief apologies,
Yes filter groups will will be fixed in phase 2 and we will be looking into the value dropdowns to implement a fix

Comment (AntoineRelief) [reply libs/shared/src/lib/components/email/filter/filter.constant.ts:278]: @HasnatL-Adapptlimited 

for the values, would the fix be for phase 1 or 2?

I really suggest you have a look at how the grid can build filters, as most of the question types should be supported

**Comment by AntoineRelief** [comment]
> @HasnatL-Adapptlimited
> 
> about this ticket: https://dev.azure.com/WHOHQ/EMSSAFE/_workitems/edit/90513 Which I believe this PR is also about
> 
> there is a difference between the dates you see in the front-end & the one sent by the back-end:
> 
> Capture d’écran 2024-04-24 à 17 31 19 Capture d’écran 2024-04-24 à 17 30 51
> I don't know which format is correct so please make according changes

@HasnatL-Adapptlimited 
do you know about this one?
That's preventing me to merge this PR

**Comment by HasnatL-Adapptlimited** [comment]
Hey @AntoineRelief,

@morgang-adappt  has been looking into this as we can't seem to reproduce this on our side

Quick Question
Is the PR with the same name being used in the backend when testing this branch?
https://github.com/ReliefApplications/ems-backend/pull/1023

**Comment by AntoineRelief** [comment]
> @HasnatL-Adapptlimited
> 
> about this ticket: https://dev.azure.com/WHOHQ/EMSSAFE/_workitems/edit/90513 Which I believe this PR is also about
> 
> there is a difference between the dates you see in the front-end & the one sent by the back-end:
> 
> Capture d’écran 2024-04-24 à 17 31 19 Capture d’écran 2024-04-24 à 17 30 51
> I don't know which format is correct so please make according changes

@HasnatL-Adapptlimited 
looking good now
I may have incorrectly done my setup indeed

**Comment by AntoineRelief** [comment]
@HasnatL-Adapptlimited 

merged back & front

cc @morgang-adappt

**Pull request merged**